app/services/currencies_service.py

```
@@ -0,0 +1,103 @@
+    async def get_all_currencies(self) -> list[dict]:
+        """
+        Get all currencies from GW2 API and sync with database in the background.
+        If API fails, fallback to database.
+        """
+        try:
+            currencies_data = await self._get_currencies_from_api()
+
+            # Sync with db in background
+            self.task = asyncio.create_task(self._sync_currencies_to_db(currencies_data))
+
+            return currencies_data
```

Ruff (0.14.1)

Consider moving this statement to an `else` block

(TRY300)